winsup/cygwin/include/cygwin/exit_process.h

```diff
@@ -71,11 +74,11 @@ get_address_from_cygwin_console_helper(int bitness, wchar_t *function_name)
 
   STARTUPINFOW si = {};
   PROCESS_INFORMATION pi;
-  size_t len = wcslen (wbuf) + wcslen (function_name) + 1;
+  size_t len = wcslen (wbuf) + 1 + wcslen (function_name) + 1 + (exit_code == 0 ? 1 : (int)(log10 (exit_code) + 1)) + 1 + (pid == 0 ? 1 : (int)(log10 (pid) + 1));
   WCHAR cmd[len + 1];
   WCHAR title[] = L"cygwin-console-helper";
 
-  swprintf (cmd, len + 1, L"%S %S", wbuf, function_name);
+  swprintf (cmd, len + 1, L"%S %S %d %d", wbuf, function_name, exit_code, pid);
```

Comment:
That second `%d` needs to be `%ld` to accommodate for a `DWORD` (which really is an `unsigned long`).

Reply:
Done https://github.com/msys2/msys2-runtime/pull/31/commits/ced32fb6d72f2dfbbf1d6026481d8c1c44742e4b